docs/source/advanced-usage.rst

```diff
@@ -69,3 +69,56 @@ directly.
 
     for event in connection.events():
         # As with WSConnection, only without any handshake events
+
+HTTP/2
+------
+
+WebSockets over HTTP/2 have a distinct difference to HTTP/1 in that
+only a single HTTP/2 stream is dedicated to the WebSocket rather than
+the entire connection (as in HTTP/1). This requires the HTTP/2
+connection to be managed before the WebSocket connection. `wsproto`
+offers two choices in this situation, it can manage either manage the
+HTTP/2 connection or not.
+
+Managed HTTP/2 Connection
+'''''''''''''''''''''''''
+
+Allowing `wsproto` to manage the HTTP/2 connection is the easiest way
+to support WebSockets over HTTP/2, however it restricts the entire
+connection to be for the WebSocket only i.e. no multiplexing of HTTP
+requests alongside the WebSocket. To use this simply provide the
+:class:`H2Handshake <wsproto.handshake.H2Handshake>` class to the
+`WSConnection` class on initialisation, as so,
+
+.. code-block:: python
+
+    from wsproto import ConnectionType, WSConnection
+    from wsproto.handshake import H2Handshake
+
+    client = WSConnection(ConnectionType.CLIENT, handshake_class=H2Handshake)
+    server = WSConnection(ConnectionType.SERVER, handshake_class=H2Handshake)
+
+the usage of the `WSConnection` instance is as with HTTP/1.
+
+Unmanaged HTTP/2 Connection
+'''''''''''''''''''''''''''
+
+If `wsproto` doesn't manage the HTTP/2 connection it can still be used
+for the WebSocket stream. The HTTP/2 connection will need to handshake
+the WebSocket stream, with the key being agreement on the extensions
+used. Once the extensions have been agreed the :class:`Connection
+<wsproto.connection.Connection>` class can be used to manage the
+WebSocket connection, noting that data to be sent or received will
+need to be parsed by the HTTP/2 connection first. In practice this
+looks like,
+
+.. code-block:: python
+
+     # WebSocket request has been received
+     request_extensions: List[str]
+     supported_extensions = [wsproto.extensions.PerMessageDeflate()]
+     accepts = handshake_extensions(request_extensions, supported_extensions)
+     if accepts:
+         response_headers.append({"sec-websocket-extensions": accepts})
+     # Send the response headers
+     connection = Connection(ConnectionType.SERVER, supported_extensions)
```

Comment:
Does the process looks different for client connections?

Reply:
Ah, good point - I've had to add a `client_extensions_handshake` and rename `handshake_extensions` to `server_extensions_handshake`.